explore/src/main/scala/explore/observationtree/ObsList.scala

```diff
@@ -60,121 +55,112 @@ object ObsList {
-                    props.focusedObs.set(focusedObs.some)
-                  )
-                )(
-                  ObsBadge(
-                    obs,
-                    selected = selected,
-                    setStatusCB = (ObsListActions
-                      .obsStatus(obs.id)
-                      .set(undoCtx) _).compose((_: ObsStatus).some).reuseAlways.some,
-                    setActiveStatusCB = (ObsListActions
-                      .obsActiveStatus(obs.id)
-                      .set(undoCtx) _).compose((_: ObsActiveStatus).some).reuseAlways.some,
-                    deleteCB = ObsListActions
-                      .obsExistence(obs.id, props.focusedObs)
-                      .mod(undoCtx)(obsListMod.delete)
-                      .reuseAlways
-                      .some
-                  )
+              )(
+                ObsBadge(
+                  obs,
+                  selected = selected,
+                  setStatusCB = (ObsListActions
+                    .obsStatus(obs.id)
+                    .set(undoCtx) _).compose((_: ObsStatus).some).reuseAlways.some,
+                  setActiveStatusCB = (ObsListActions
+                    .obsActiveStatus(obs.id)
+                    .set(undoCtx) _).compose((_: ObsActiveStatus).some).reuseAlways.some,
+                  deleteCB = ObsListActions
+                    .obsExistence(obs.id, props.focusedObs)
+                    .mod(undoCtx)(obsListMod.delete)
+                    .reuseAlways
+                    .some
                 )
-              }
-            )
+              )
+            }
           )
         )
-      }
-  }
+      )
+    }
 
   protected val component =
-    ScalaComponent
-      .builder[Props]
-      .renderBackend[Backend]
-      .componentDidMount { $ =>
-        val observations = $.props.observations.get
-
-        // If focused observation does not exist anymore, then unfocus.
-        $.props.focusedObs.mod(_.flatMap {
-          case oid if !observations.contains(oid) => none
-          case other                              => other.some
-        })
-      }
-      .componentDidUpdate { $ =>
-        val prevObservations = $.prevProps.observations.get
-        val observations     = $.currentProps.observations.get
-
-        // If focused observation does not exist anymore, then focus on closest one.
-        $.currentProps.focusedObs.mod(_.flatMap {
-          case oid if !observations.contains(oid) =>
-            prevObservations
-              .getIndex(oid)
-              .flatMap { idx =>
-                observations.toList.get(math.min(idx, observations.length - 1).toLong)
-              }
-              .map(_.id)
-          case other                              => other.some
-        })
+    ScalaFnComponent
+      .withHooks[Props]
+      // Saved index into the observation list
+      .useState(none[Int])
+      .useEffectWithDepsBy((props, _) => (props.focusedObs, props.observations)) {
```

Comment:
This replaces both componentDidMount and componentDidUpdate. The logic is quite different, but it seems to be working properly...